rspace/src/test/scala/coop/rchain/rspace/history/CachingHistorySpec.scala

```diff
@@ -0,0 +1,60 @@
+package coop.rchain.rspace.history
+
+import coop.rchain.rspace.history.HistoryInstances.{CachingHistoryStore, MergingHistory}
+import coop.rchain.rspace.history.TestData._
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.{FlatSpec, Matchers, OptionValues}
+
+import scala.concurrent.duration._
+
+class CachingHistorySpec
+    extends FlatSpec
+    with Matchers
+    with OptionValues
+    with InMemoryHistoryTestBase {
+
+  def create: (MergingHistory[Task], CachingHistoryStore[Task]) = {
+    val historyStore = inMemHistoryStore
+    val caching      = CachingHistoryStore(historyStore)
+    val history      = new MergingHistory[Task](History.emptyRootHash, caching)
+    (history, caching)
+  }
+
+  protected def withEmptyTrie(
+      f: (MergingHistory[Task], CachingHistoryStore[Task]) => Task[Unit]
+  ): Unit = {
+    val (history, caching) = create
+    f(history, caching).flatMap(_ => history.close()).runSyncUnsafe(20.seconds)
```

Comment:
Could you extract `runSyncUnsafe` to a common place for all the tests? Or at least for this one and others could be updated on demand